src/app/components/AutoComplete.svelte

```diff
@@ -0,0 +1,188 @@
+<script lang="ts">
+  // TODO: the following interface should be removed once #33 is done
+  interface dataObj {
+    bcp47Tag: string;
+    language: string;
+  }
+  enum keyboardKey {
+    Up = "ArrowUp",
+    Down = "ArrowDown",
+    Enter = "Enter",
+  }
+  export let label = "";
+  export let subtext = "";
+  // Testing Data
+  export let results: dataObj[] = [];
+  // To store Filted array
+  let filetred: dataObj[] = [];
+  // To store selected language
+  let selected: string = "";
+  // Toggle to show search list
+  let show = false;
+  // Index of Previous focus element
+  let prevIndex = -1;
+  // Index of focus element
+  let index = -1;
+
+  // Input to search
+  function onChange(event: any) {
+    show = true;
+    filetred = results.filter((item: any) => {
+      let a = new RegExp("^" + event.target.value.toUpperCase());
+      return a.test(item.language.toUpperCase());
+    });
+  }
+
+  function closeSuggestion() {
+    show = false;
+  }
+
+  // If anywhere on the screen is clicked
+  // Derived from https://github.com/rster2002/svelte-outside-click
+  function clickOutside(node: HTMLUListElement, onEventFunction: Function) {
+    const handleClick = (event: MouseEvent) => {
+      const path = event.composedPath();
+      if (!path.includes(node)) {
+        onEventFunction();
+      }
+    };
+    document.addEventListener("click", handleClick);
+    return {
+      destroy() {
+        document.removeEventListener("click", handleClick);
+      },
+    };
+  }
+
+  // On select item in list
+  function selectedList(data: dataObj) {
+    show = false;
+    selected = data.language;
+    subtext = data.bcp47Tag;
+  }
+
+  const mod = (a: number, n: number) => {
+    return a - n * Math.floor(a / n);
+  };
```

Comment:
Why is this used instead of `a % n`?

Reply:
javascript seems like it not smart enough to handle modulo with negative numerator, this function is to handle modules with both positive and negative numerator

Reply:
Ah, when index becomes `-1` `-1 % allItems.length === -1` where as `mod(-1, allItems.length) === allItems.length -1`. The latter is what we want! 👍🏼